src/test/java/seedu/address/model/step/UniqueStepListTest.java

```diff
@@ -0,0 +1,150 @@
+        uniqueStepList.add(TESTSTEP);
+        uniqueStepList.setStep(TESTSTEP, TESTSTEP2);
+        UniqueStepList expectedUniqueStepList = new UniqueStepList();
+        expectedUniqueStepList.add(TESTSTEP2);
+        assertEquals(expectedUniqueStepList, uniqueStepList);
+    }
+
+    @Test
+    public void setStep_editedStepHasNonUniqueDescription_throwsDuplicateStepException() {
+        uniqueStepList.add(TESTSTEP);
+        uniqueStepList.add(TESTSTEP2);
+        assertThrows(DuplicateStepException.class, () -> uniqueStepList.setStep(TESTSTEP, TESTSTEP2));
+    }
+
+    @Test
+    public void remove_nullStep_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueStepList.remove(null));
+    }
+
+    @Test
+    public void remove_StepDoesNotExist_throwsStepNotFoundException() {
+        assertThrows(StepNotFoundException.class, () -> uniqueStepList.remove(TESTSTEP));
+    }
+
+    @Test
+    public void remove_existingStep_removesStep() {
+        uniqueStepList.add(TESTSTEP);
+        uniqueStepList.remove(TESTSTEP);
+        UniqueStepList expectedUniqueStepList = new UniqueStepList();
+        assertEquals(expectedUniqueStepList, uniqueStepList);
+    }
+
+    @Test
+    public void setSteps_nullUniqueStepList_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueStepList.setSteps((UniqueStepList) null));
+    }
+
+    @Test
+    public void setSteps_uniqueStepList_replacesOwnListWithProvidedUniqueStepList() {
+        uniqueStepList.add(TESTSTEP);
+        UniqueStepList expectedUniqueStepList = new UniqueStepList();
+        expectedUniqueStepList.add(TESTSTEP2);
+        uniqueStepList.setSteps(expectedUniqueStepList);
+        assertEquals(expectedUniqueStepList, uniqueStepList);
+    }
+
+    @Test
+    public void setSteps_nullList_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> uniqueStepList.setSteps((List<Step>) null));
+    }
+
+    @Test
+    public void setSteps_list_replacesOwnListWithProvidedList() {
+        uniqueStepList.add(TESTSTEP);
+        List<Step> stepList = Collections.singletonList(TESTSTEP2);
+        uniqueStepList.setSteps(stepList);
+        UniqueStepList expectedUniqueStepList = new UniqueStepList();
+        expectedUniqueStepList.add(TESTSTEP2);
+        assertEquals(expectedUniqueStepList, uniqueStepList);
+    }
+
+    @Test
+    public void setSteps_listWithDuplicateSteps_throwsDuplicateStepException() {
+        List<Step> listWithDuplicateSteps = Arrays.asList(TESTSTEP, TESTSTEP);
+        assertThrows(DuplicateStepException.class, () -> uniqueStepList.setSteps(listWithDuplicateSteps));
+    }
+
+    @Test
+    public void asUnmodifiableObservableList_modifyList_throwsUnsupportedOperationException() {
+        assertThrows(UnsupportedOperationException.class, ()
+                -> uniqueStepList.asUnmodifiableObservableList().remove(0));
```

Comment:
The indentation level should be shifted one tab to the left

Reply:
which one? I think the checkstyle requires two tabs for line wrapping

Reply:
Line 148, says lambda is at 16, but should be at 12

Reply:
oh yeah okay

Reply:
fixed